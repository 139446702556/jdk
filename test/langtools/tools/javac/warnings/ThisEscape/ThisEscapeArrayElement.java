/*
 * @test /nodynamiccopyright/
 * @bug 8015831
 * @compile/ref=ThisEscapeArrayElement.out -Xlint:this-escape -XDrawDiagnostics ThisEscapeArrayElement.java
 * @summary Verify 'this' escape detection can follow references embedded as array elements
 */

public class ThisEscapeArrayElement {

    public ThisEscapeArrayElement() {
        final Object[][] array = new Object[][] { { this } };
        ((ThisEscapeArrayElement)array[0][0]).mightLeak();
    }

    public void mightLeak() {
    }
}
