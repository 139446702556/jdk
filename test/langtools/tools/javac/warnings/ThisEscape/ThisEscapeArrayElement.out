ThisEscapeArrayElement.java:12:56: compiler.warn.possible.this.escape
1 warning
